src/oop/object_interactions/medium/q6/MainTest.java

```diff
@@ -0,0 +1,191 @@
+            testClass.callMethod(setNumPassengersMethodName, new Object[][]{{updatePass, int.class}}, classInstance);
+            _assertEquals(updatePass, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "You have modified the provided " + setNumPassengersMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+        }
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(20, 10, 40, 35),
+                Arguments.of(10, 20, 15, 30),
+                Arguments.of(150, 200, 100, 50),
+                Arguments.of(20, 20, 10, 20)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctTransferPassengersMethod(int seat, int pass, int busySeat, int busyPass) throws Throwable {
+        Object[][] arguments = {
+                {seat, int.class},
+                {pass, int.class}
+        };
+        Object[][] busyArguments = {
+                {busySeat, int.class},
+                {busyPass, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object busyInstance = testClass.createInstance(busyArguments);
+        Object[][] methodArguments = {
+                {busyInstance, Airplane.class}
+        };
+        Object output = null;
+        if (seat - pass > 0) {
+            int move = Math.min(busyPass - busySeat, seat - pass);
+            output = testClass.callMethod(transferPassengersMethodName, methodArguments, new String[]{"public"}, classInstance, new Clause[]{
+                    new StringLiteral("We moved " + move + " passengers"),
+                    new Optional(new StringLiteral(" "))
+            });
+            _assertEquals(pass + move, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "Your " + transferPassengersMethodName + " method does not correctly add the " + numPassengersAttributeName + " attribute of the recipient.");
+            _assertEquals(busyPass - move, testClass.getFieldValue(busyInstance, numPassengersAttributeName),
+                    "Your " + transferPassengersMethodName + " method does not correctly decrease the " + numPassengersAttributeName + " attribute of the donor.");
+        } else {
+            output = testClass.callMethod(transferPassengersMethodName, methodArguments, new String[]{"public"}, classInstance, new Clause[]{
+                    new StringLiteral("This plane is full too and cannot accommodate anymore passengers"),
+                    new Optional(new StringLiteral(" "))
+            });
+            _assertEquals(pass, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "Your " + transferPassengersMethodName + " method does not correctly increase the " + numPassengersAttributeName + " attribute of the recipient.");
```

Comment:
these error messages do not make sense because we aren't testing if the passenger counts are increasing or decreasing, we are testing that they stay the same

Reply:
They also read like they are from a transplant question